part2/index.rb

```diff
@@ -0,0 +1,152 @@
+class Station
+  attr_reader :trains, :name
+
+  def initialize(name)
+    @name = name
+    @trains = []
+  end
+
+  # прием поезда
+  def add_train(train)
+    @trains << train
+  end
+
+  # кол-во поездов на станции по типу
+  def trains_by_type(type)
+    @trains.select{|train| train.type == type}.length 
+  end
+
+  # отправка поезда
+  def launch_train(train)
+    trains.include?(train) ? trains.delete(train) : (puts "Данного поезда нет на станции.")
+  end 
+
+end
+
+class Route
+  attr_reader :stations
+
+  def initialize(start_station, finish_station)
+    @stations = [start_station, finish_station]
+  end
+
+  def first_station
+    @stations.first
+  end
+
+  def finish_station
+    @stations.last
+  end
+
+  #добавить станцию в промежуточный маршрут
+  def add_station(station)
+    if !@stations.include?(station)
+      @stations.insert(-2, station)
+    end
+  end
+
+  #удалить станцию из промежуточного маршрута
+  def remove_station(station)
+    if (@stations.include?(station) && (station != @stations.first) && (station != @stations.last))
```

Comment:
```suggestion
    if station != @stations.first) && station != @stations.last
```
проверка на наличие скорее лишняя
скобки не пишут обычно в таких случаях, т.к. операторы `&&`, `||` имеют более высокий приоритет, чем операторы сравнения